src/org/omegat/filters2/text/ini/INIFilter.java

```diff
@@ -107,37 +114,47 @@ public void processFile(BufferedReader reader, BufferedWriter outfile, FilterCon
                 continue;
             }
 
+            // retrieve section as group name
             if (trimmed.startsWith("[") && trimmed.endsWith("]")) {
-                // group name
                 group = trimmed.substring(trimmed.offsetByCodePoints(0, 1),
                         trimmed.offsetByCodePoints(trimmed.length(), -1));
+                outfile.write(str);
+                outfile.write(lbpr.getLinebreak());
+                continue;
             }
 
             // key=value pairs
-            int equalsPos = str.indexOf('=');
+            equalsPos = str.indexOf('=');
 
-            // if there's no separator, assume it's a key w/o a value
+            // if there's no separator, assume it's an unnamed key with a value
+            // It may be a continuous line, so allow translate.
             if (equalsPos == -1) {
-                equalsPos = str.offsetByCodePoints(str.length(), -1);
-            }
-
-            // advance if there're spaces after =
-            while (str.codePointCount(equalsPos, str.length()) > 1) {
-                int nextOffset = str.offsetByCodePoints(equalsPos, 1);
-                if (str.codePointAt(nextOffset) != ' ') {
-                    break;
+                // produce virtual key
+                key = (group != null ? group + '/' : "") + String.format(unnamed_format, unnamed_counter);
```

Comment:
Three things:

1. Sorry for the bikeshedding, but can we use camelCase for `unnamed_format` and `unnamed_counter`?

2. `String#format` is surprisingly expensive; since we only have one use of `unnamed_format`, I would suggest instead making a builder method like

    ```java
    private String buildKey(String group, int counter) {
        StringBuilder sb = new StringBuilder();
        if (group != null) {
            sb.append(group).append('/');
        }
        sb.append("[$$UnNamee_").append(counter).append("_]");
        return sb.toString();
    }
    ```
3. Is `[$$UnNamee_%d_]` really the intended key? Not UnName*****d*****? Is there some significance of this particular string, or is it something arbitrary you decided?

Reply:
Fixed comments No.1 and No.2